src/js/variant-widget/eva-variant-widget.js

```diff
@@ -304,7 +304,7 @@ EvaVariantWidget.prototype = {
                             var groupedArr = _.groupBy(tempArray);
                             var so_array = [];
                             _.each(_.keys(groupedArr), function (key) {
-                                var index = _.indexOf(consequenceTypesHierarchy, key);
+                                var index = _.indexOf(_.keys(consequenceTypeDetails), key);
```

Comment:
This function is repeated several times across this file. In some cases there is an if-else conditional and in other 2 there isn't one. Is there a reason for that? Could you please extract the block that builds `so_array` to a function so it can be reused?